cobbler/actions/reposync.py

```diff
@@ -570,12 +573,16 @@ def yum_sync(self, repo):
             proxy = repo.proxy
         (cert, verify) = self.gen_urlgrab_ssl_opts(repo.yumopts)
 
-        # FIXME: These two variables were deleted
-        repodata_path = ""
-        repomd_path = ""
+        repodata_path = os.path.join(temp_path, "repodata")
```

Comment:
```suggestion
        repodata_path = os.path.join(dest_path, "repodata")
```

See https://github.com/cobbler/cobbler/pull/2942/files#diff-77204e56ea1a781e3396e07cc7dea840201908389aa0a4bb0e00a6752d2a1e68R512

Reply:
I made the requested changes.

Reply:
Running more local introspective testing this morning I'm getting this:
```
received on stderr: 
Deleted old repo metadata for /var/www/cobbler/repo_mirror/eog-local8-testing/.origin/repodata
Exception occurred: <class 'OSError'>
Exception value: [Errno unknown] Cannot open /var/www/cobbler/repo_mirror/eog-local8-testing/.origin/repodata/repomd.xml: No such file or directory: 'Input/Output error'
Exception Info:
  File "/usr/lib/python3.6/site-packages/cobbler/actions/reposync.py", line 161, in run
    self.sync(repo)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/reposync.py", line 204, in sync
    self.yum_sync(repo)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/reposync.py", line 597, in yum_sync
    h.perform(r)

  File "/usr/lib64/python3.6/site-packages/librepo/__init__.py", line 1595, in perform
    _librepo.Handle.perform(self, result)
```
Reverting this change results in successful task completion:
```
[root@default-centos8-qemu ~]# cobbler reposync --only=eog-local8-testing
task started: 2022-08-18_101136_reposync
task started (id=Reposync, time=Thu Aug 18 10:11:36 2022)
running python triggers from /var/lib/cobbler/triggers/task/reposync/pre/*
running shell triggers from /var/lib/cobbler/triggers/task/reposync/pre/*
shell triggers finished successfully
hello, reposync
run, reposync, run!
creating: /var/www/cobbler/repo_mirror/eog-local8-testing/config.repo
creating: /var/www/cobbler/repo_mirror/eog-local8-testing/.origin/eog-local8-testing.repo
running: /usr/bin/dnf reposync --newest-only --delete --remote-time --config=/var/www/cobbler/repo_mirror/eog-local8-testing/.origin/eog-local8-testing.repo --repoid=eog-local8-testing -p /var/www/cobbler/repo_mirror -a x86_64 -a noarch
3.0 kB     00:00    
[SKIPPED] actions-runner-2.285.1-3.eog.el8.x86_64.rpm: Already downloaded      
[SKIPPED] eog-devops-certs-3.0-1.eog.el8.noarch.rpm: Already downloaded        
[SKIPPED] epkg-minimal-1.2.0-1.el8.noarch.rpm: Already downloaded              
[SKIPPED] github-backup-utils-3.4.0-3.eog.el8.noarch.rpm: Already downloaded   
[SKIPPED] hypr-radius-7.2.0-1.eog.el8.noarch.rpm: Already downloaded           
[SKIPPED] insight-platform-collector-2.0.2-2.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] koji-eog-builder-plugins-3.4-1.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] koji-eog-hub-plugins-3.4-1.eog.el8.noarch.rpm: Already downloaded    
[SKIPPED] koji-hub-plugins-filesystem-1.1-1.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] koji-hub-plugins-reposync-1.0-1.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] koji-hub-plugins-tag2distrepo-2-1.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] painless-password-rotation-0^20200128git8387611-2.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] painless-password-rotation-template-0^20200128git8387611-2.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] phpipam-httpd-conf-1.5.0-6.eog.el8.noarch.rpm: Already downloaded    
[SKIPPED] phpipam-langpack-i18n-1.5.0-6.eog.el8.noarch.rpm: Already downloaded 
[SKIPPED] phpipam-server-1.5.0-6.eog.el8.noarch.rpm: Already downloaded        
[SKIPPED] rabbitmq-message-deduplication-0.4.4-4.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] saltdk-2021.9.3-3.eog.el8.noarch.rpm: Already downloaded             
[SKIPPED] tinystatus-0^20210709gitfc128ad-5.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] tinystatus-httpd-conf-0^20210709gitfc128ad-5.eog.el8.noarch.rpm: Already downloaded

received on stderr: 
running: createrepo  -c cache -s sha /var/www/cobbler/repo_mirror/eog-local8-testing
received on stdout: Directory walk started
Directory walk done - 20 packages
Temporary output repo path: /var/www/cobbler/repo_mirror/eog-local8-testing/.repodata/
Preparing sqlite DBs
Pool started (with 5 workers)
Pool finished

received on stderr: 
running: chown -R root:apache /var/www/cobbler/repo_mirror/eog-local8-testing
received on stdout: 
received on stderr: 
running: chmod -R 755 /var/www/cobbler/repo_mirror/eog-local8-testing
received on stdout: 
received on stderr: 
*** TASK COMPLETE ***
```

I will push my changes back into this PR.

Reply:
@c4t3l Is it possible that you check if this is then a problem on our development version? Because if this works on `main`, I am confused why it doesn't work here...

Reply:
I checked out main (9d35aa29ab3ee19598dc16d9c1e8c70eac947d44) and built the rpm package (for epel8 in my case).  Please see output below:
```
[root@default-centos8-qemu ~]# cobbler reposync --only=eog-local8-testing
task started: 2022-08-18_154407_reposync
task started (id=Reposync, time=Thu Aug 18 15:44:07 2022)
reposync; ['eog-local8-testing']
hello, reposync
run, reposync, run!
running: /usr/bin/dnf reposync --newest-only --delete --refresh --remote-time --config=/var/www/cobbler/repo_mirror/eog-local8-testing/.origin/eog-local8-testing.repo --repoid=eog-local8-testing -p /var/www/cobbler/repo_mirror -a x86_64
190 kB     00:00    
running: createrepo  -c cache -s sha /var/www/cobbler/repo_mirror/eog-local8-testing
received on stdout: Directory walk started
running: chown -R root:apache /var/www/cobbler/repo_mirror/eog-local8-testing
received on stdout: 
running: chmod -R 755 /var/www/cobbler/repo_mirror/eog-local8-testing
received on stdout: 
*** TASK COMPLETE ***
```

Cobbler log output:
```
[Thread-4] 2022-08-18T15:44:07 - INFO | REMOTE start_task(Reposync); event_id(2022-08-18_154407_reposync); user(?)
[2022-08-18_154407_reposync] 2022-08-18T15:44:08 - INFO | reposync; ['eog-local8-testing']
[2022-08-18_154407_reposync] 2022-08-18T15:44:08 - INFO | hello, reposync
[2022-08-18_154407_reposync] 2022-08-18T15:44:08 - INFO | run, reposync, run!
[2022-08-18_154407_reposync] 2022-08-18T15:44:08 - INFO | running: /usr/bin/dnf reposync --newest-only --delete --refresh --remote-time --config=/var/www/cobbler/repo_mirror/eog-local8-testing/.origin/eog-local8-testing.repo --repoid=eog-local8-testing -p /var/www/cobbler/repo_mirror -a x86_64
[2022-08-18_154407_reposync] 2022-08-18T15:44:14 - INFO | received on stdout: eog-local8-testing                              785 kB/s | 190 kB     00:00    
actions-runner-2.285.1-3.eog.el8.x86_64.rpm      49 MB/s | 104 MB     00:02    

[2022-08-18_154407_reposync] 2022-08-18T15:44:14 - INFO | running: createrepo  -c cache -s sha /var/www/cobbler/repo_mirror/eog-local8-testing
[2022-08-18_154407_reposync] 2022-08-18T15:44:16 - INFO | received on stdout: Directory walk started
Directory walk done - 1 packages
Temporary output repo path: /var/www/cobbler/repo_mirror/eog-local8-testing/.repodata/
Preparing sqlite DBs
Pool started (with 5 workers)
Pool finished

[2022-08-18_154407_reposync] 2022-08-18T15:44:16 - INFO | running: chown -R root:apache /var/www/cobbler/repo_mirror/eog-local8-testing
[2022-08-18_154407_reposync] 2022-08-18T15:44:16 - INFO | received on stdout: 
[2022-08-18_154407_reposync] 2022-08-18T15:44:16 - INFO | running: chmod -R 755 /var/www/cobbler/repo_mirror/eog-local8-testing
[2022-08-18_154407_reposync] 2022-08-18T15:44:16 - INFO | received on stdout: 
[2022-08-18_154407_reposync] 2022-08-18T15:44:16 - INFO | ### TASK COMPLETE ###
```

So in this instance it worked, but only pulled down one x86_64 package.  I guess next step would be firing up a python debugger to see why the `dest_path` variable fails in v3.2.2?

Reply:
Nah don't bother for the validation of this PR. This PR is now fine. If you desire to investigate then feel free to do so.